src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -471,31 +475,30 @@ private bool CheckMarkers(BuildUpToDateCheckLogger logger, IDictionary<string, D
 
         private bool CheckCopyToOutputDirectoryFiles(BuildUpToDateCheckLogger logger, IDictionary<string, DateTime> timestampCache)
         {
-            var items = _items.SelectMany(kvp => kvp.Value).Where(item => item.CopyType == CopyToOutputDirectoryType.CopyIfNewer)
-                .Select(item => item.Path);
+            var items = _items.SelectMany(kvp => kvp.Value).Where(item => item.CopyType == CopyToOutputDirectoryType.CopyIfNewer);
 
             string outputFullPath = Path.Combine(_msBuildProjectDirectory, _outputRelativeOrFullPath);
 
             foreach (var item in items)
             {
-                var filename = Path.GetFileName(item);
+                var filename = item.Link ?? Path.GetFileName(item.Path);
```

Comment:
@Pilchie At this point we are checking CopyToOutputDirectory files for up to dateness. So now we check to see if an item has Link metadata and use that if it exists. Otherwise, we fall back to the filename of the item.

Reply:
Does this issue only affect linked files that are *also* copy to output?  I would have expected to need a change somewhere else for linked source files.

Reply:
In this case, yes, I believe so. In the case of a file that is just an input to compilation and is not copied to the output, we only need to look at the timestamp on the original file. So we want to look at the original path of the item in that case (which we do, elsewhere), we don't care what the link name is. If that linked file is copied, however, then we need to follow the target name, not the original name.